content/articles/creating-an-animated-javafx-field-validation-alert/index.md

```diff
@@ -0,0 +1,758 @@
+
+In this article, you shall be able to do some input validations, effects, and animations based on the outcome on the form.
+
+### Table of Content
+
+- [Key takeaways](#key-takeaways)
+- [Pre-requisites](#pre-requisites)
+- [Create a new project](#create-a-new-project)
+    - [Folder Structure](#folder-structure)
+- [Create the Login form](#create-the-login-form)
+    - [Modify the Scene](#modify-the-scene)
+    - [Design the view](#design-the-view)
+- [Add some validations to the form](#add-some-validations-to-the-form)
+- [Add animations to the project](#add-animations-to-the-project)
+- [Further learning](#further-learning)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### Key takeaways
+
+At the end of the article, you will have done the following:
+
+- Create a JavaFX login form
+- Apply styling to the application
+- Create an input validation method
+- Add effects based on the outcomes of the validation
+- Add animations on the controls
+
+### Pre-requisites
+
+The basics needed to follow up with the tutorial quickly include:
+
+- Must be familiar with Java language
+- Java SDK installed on the machine
+- A JavaFX IDE is installed on the machine.
+  I recommend using the latest IntelliJ IDEA Community version. It is free and enables developers to quickly create a JavaFX project.
+  Find it in this [link](https://www.jetbrains.com/idea/download).
+  The latest Ultimate version also supports JavaFX projects.
+- A good internet connection. This will be used in installing additional dependencies.
+
+Once all the above pre-requisites are met, follow the steps below to accomplish the goals:
+
+> The Screenshot images in the article are of the IntelliJ Ultimate edition `2021.2.2` version
+
+### Create a new project
+
+- Open the IDE and click on create a new project. Under the project type options, select JavaFX.
+  Set the following on the right-hand side window:
+
+- `Name`: Login
+- `Group`: authentication
+- `Artifact`: Login
+- `Project SDK`: You can set the latest version or that which you are using.
+
+These are well shown in the image below:
+
+![new javafx project](/engineering-education/creating-an-animated-javafx-field-validation-alert/new-javafx-project.png "New-JavaFX project")
+
+- Click on the `Next` button. On the dependencies window, select the `BootstrapFX`, `ControlsFX`, and `FormsFX` checkboxes.
+  BootstrapFX helps in styling the form using JavaFX styling. The styling is somehow similar to CSS styling.
+  ControlsFX is for allowing controls to be available in the project.
+  These include TextFields, PasswordFields, Labels, DatePickers, and many others. FormFX allows one to easily design forms.
+

```

Comment:
To communicate full ideas. Organise your content into paragraphs with a single one communicating a single idea.

Reply:
Got it!